docs/developer/tips.md

```diff
@@ -0,0 +1,12 @@
+# Developer tips and best practices
+
+## Using Providers for MyST content
+
+When using components that render MyST content (using `<MyST ast={content} />`), they must be wrapped wrap them in various providers e.g. `TabStateProvider` to ensure proper React context is available.
```

Comment:
I think this has little to do with `MyST` components. It might be something like:

`Providers` create [React contexts](https://react.dev/reference/react/createContext) for communicating app states across components. For example,
the `<ArticleProvider>...</ArticleProvider>` shares an `ArticleContext` with all its children, telling them the current article `kind`, `frontmatter`, and `references`.

Reply:
This one still, then merge?